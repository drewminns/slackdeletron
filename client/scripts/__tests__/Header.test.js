import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Header from '../Components/Header';

const AVATAR =
  'https://secure.gravatar.com/avatar/f2d55f35afbc3f607e34504f5a2d7d88.jpg?s=192&d=https%3A%2F%2Fa.slack-edge.com%2F7fa9%2Fimg%2Favatars%2Fava_0006-192.png';
const NAME = 'George Jetson';

describe('<Header />', () => {
  it('Renders Correctly', () => {
    const header = renderer.create(<Header />).toJSON();
    expect(header).toMatchSnapshot();
  });

  it('renders a Header component', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toHaveLength(1);
  });

  it('renders a Header component in the loggedOut State', () => {
    const wrapper = mount(<Header />);
    expect(wrapper.find('a').text()).toBe('Login with Slack');
  });

  it('renders a loggedIn Header with a name and an avatar', () => {
    const wrapper = mount(<Header name={NAME} avatar={AVATAR} isLoggedIn />);
    const text = wrapper.find('p');
    const image = wrapper.find('img');
    const link = wrapper.find('a');
    expect(link.text()).toBe('Logout');
    expect(text.text()).toBe(`Hey ${NAME}`);
    expect(image.prop('src')).toEqual(AVATAR);
    expect(image.prop('alt')).toEqual(NAME);
  });

  it('does not render a Header component in the loggedIn State with no other props provided', () => {
    const wrapper = mount(<Header isLoggedIn />);
    expect(wrapper.find('a').text()).toBe('Login with Slack');
  });

  it('does not render a Header component in the loggedIn State when other values are provided', () => {
    const wrapper = mount(<Header name={NAME} avatar={AVATAR} />);
    const linkText = wrapper.find('a');
    const text = wrapper.find('p');
    const image = wrapper.find('img');
    expect(linkText.text()).toBe('Login with Slack');
    expect(text).toHaveLength(0);
    expect(image).toHaveLength(0);
  });
});
