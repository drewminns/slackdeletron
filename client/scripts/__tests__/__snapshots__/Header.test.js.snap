// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Renders Correctly 1`] = `
<header
  className="sc-EHOje kKNruh"
>
  <div
    className="sc-bwzfXH pjOvZ"
  >
    <div
      className="sc-bwzfXH fTUftp"
    >
      <h1
        className="sc-ifAKCX dLitqJ"
      >
        Slack Deletron
      </h1>
    </div>
    <div
      className="sc-bwzfXH jJbKFp"
    >
      <a
        className="sc-bxivhb fRTnxb"
        href="api/slack/login"
      >
        Login with Slack
      </a>
    </div>
  </div>
</header>
`;
